falcon_generate_changelist.py

```diff
@@ -0,0 +1,148 @@
+        return output
+    except:
+        traceback.print_exc()
+        cleanup()
+        sys.exit(1)
+
+
+def append_to_changelist(file_name, string):
+    with open(file_name, "a") as f:
+        print >> f, string
+
+
+def cleanup(ask=True):
+    if ask is True:
+        print "OK to delete temporary and log files? (y/N): "
+        response = raw_input()
+    if ask is False or (ask is True and response == "y"):
+        tmp_files = glob.glob(os.path.join(TMP_CHANGELIST + "*"))
+        for f in tmp_files:
+            os.remove(f)
+        if os.path.isfile(OLD_CHANGELIST):
+            os.remove(OLD_CHANGELIST)
+        LOG_FILE.close()
+        os.remove(LOG_FILE_NAME)
+
+
+print "Generating new %s for FALCON release %s" % (CHANGELIST, NEW_RELEASE_VERSION)
+if os.path.isfile(TMP_CHANGELIST):
+    os.remove(TMP_CHANGELIST)
+if os.path.isfile(OLD_CHANGELIST):
+    os.remove(OLD_CHANGELIST)
+
+print "Getting commits between tag %s and HEAD" % PREV_RELEASE_GIT_TAG
+hashes = run_cmd("git log %s..HEAD --pretty='%%h'" % PREV_RELEASE_GIT_TAG).split()
+
+print "Getting details of %s commits" % len(hashes)
+
+for h in hashes:
+    subject = run_cmd("git log %s -1 --pretty='%%s' | head -1" % h).strip()
+    body = run_cmd("git log %s -1 --pretty='%%b'" % h)
+    committer = run_cmd("git log %s -1 --pretty='%%cn <%%ce>' | head -1" % h).strip()
+    body_lines = body.split("\n")
+
+    match = re.search("\[(\w+)\]" , subject)
+    # Default category
+    category = "BUG"
+    if match:
+        category = match.group(1).upper()
+        print "Subject : %s, Category : %s " % (subject, category)
+        subject = re.sub("\[(\w+)\]", "", subject)
+    file_name = TMP_CHANGELIST + "_" + category
+    if not (os.path.isfile(file_name)):
+        append_to_changelist(file_name, category+"S")
+        append_to_changelist(file_name, "=======================================")
+    if "Merge pull" in subject:
+        # Parse old format commit message
+        append_to_changelist(file_name, "  %s" % subject)
+        append_to_changelist(file_name, "  [%s]" % body_lines[0])
+        append_to_changelist(file_name, "")
+
+    elif "maven-release" not in subject:
+        # Parse new format commit message
+        # Get authors from commit message, committer otherwise
+        authors = [committer]
+        if "Author:" in body:
+            authors = [line.split(":")[1].strip() for line in body_lines if "Author:" in line]
+
+        # Generate GitHub PR URL for easy access if possible
+        github_url = ""
+        if "Closes #" in body:
+            pr_num = [line.split()[1].lstrip("#") for line in body_lines if "Closes #" in line][0]
+            github_url = "github.com/apache/falcon/pull/%s" % pr_num
+
+        append_to_changelist(file_name, "  %s" % subject)
+        append_to_changelist(file_name, "  %s" % ', '.join(authors))
+        # for author in authors:
+        #     append_to_changelist("  %s" % author)
```

Comment:
That is true. Now the commit can be easily traced back to the contributor. So, this may not be really necessary. Was just keeping it inline with our previous CHANGES.txt and also since the falcon_merge_pr.py allows for multiple authors, if there is more than one author, thought it will get called out here too.

No personal preference. Can knock if off.